packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -3976,23 +3952,22 @@ class InheritedElement extends ProxyElement {
     super.debugDeactivated();
   }
 
+  /// Calls [Element.didChangeDependencies] of all dependent elements, if
+  /// [InheritedWidget.updateShouldNotify] returns true.
+  ///
+  /// Notifies all dependent elements that this inherited widget has changed.
+  ///
+  /// [InheritedElement] calls this function if the [widget]'s
+  /// [InheritedWidget.updateShouldNotify] returns true.
+  ///
+  /// This method must be called during the build phase. Usually this method is
+  /// called automatically when an inherited widget is rebuilt, e.g. as a
+  /// result of calling [State.setState] above the inherited widget.
   @override
   void notifyClients(InheritedWidget oldWidget) {
     if (!widget.updateShouldNotify(oldWidget))
       return;
-    dispatchDidChangeDependencies();
-  }
-
-  /// Notifies all dependent elements that this inherited widget has changed.
-  ///
-  /// [InheritedElement] calls this function if [InheritedWidget.updateShouldNotify]
-  /// returns true. Subclasses of [InheritedElement] might wish to call this
-  /// function at other times if their inherited information changes outside of
-  /// the build phase. [InheritedWidget] subclasses can also call this directly
-  /// by first obtaining their [InheritedElement] using
-  /// [BuildContext.ancestorInheritedElementForWidgetOfExactType].
-  void dispatchDidChangeDependencies() {
```

Comment:
this change seems fine. We should announce it on flutter-dev though, since it's a breaking change.

Reply:
Yep, posted.